src/libsync/owncloudpropagator.cpp

```diff
@@ -939,6 +940,11 @@ void PropagateDirectory::slotSubJobsFinished(SyncFileItem::Status status)
     emit finished(status);
 }
 
+void PropagateDirectory::slotAbortFinished()
+{
+    emit abortFinished();
```

Comment:
You need to count the jobs and only emit the signal when ALL subjobs have finished their abort.